File: leetcode/0058_length_of_last_word.kt
package leetcode

class Solution {
    fun lengthOfLastWord(s: String): Int {
        var j = s.length - 1
        while (j >= 0 && s[j] == ' ')
            j--

        var i = j
        while (i >= 0 && s[i] != ' ')
            i--

        return j - i
    }
}

fun main(args: Array<String>) {
    val solution = Solution()

    assert(5 == solution.lengthOfLastWord("Hello World"))
    assert(1 == solution.lengthOfLastWord("a"))
    assert(1 == solution.lengthOfLastWord("a "))
}

File: leetcode/0058_length_of_last_word.py
# -*- coding: utf-8 -*-


class Solution:
    def lengthOfLastWord(self, s):
        j = len(s) - 1
        while j >= 0 and s[j] == ' ':
            j -= 1

        i = j
        while i >= 0 and s[i] != ' ':
            i -= 1

        return j - i


if __name__ == '__main__':
    solution = Solution()

    assert 5 == solution.lengthOfLastWord('Hello World')
    assert 1 == solution.lengthOfLastWord('a')
    assert 1 == solution.lengthOfLastWord('a ')
